src/input/type_record.cc

```diff
@@ -617,10 +569,9 @@ const Record  & AbstractRecord::get_descendant(unsigned int idx) const
 
 
 const Record * AbstractRecord::get_default_descendant() const {
-    if (data_->keys.size() != 0 )  { // skip for empty records
-        Default &dflt = data_->keys[0].default_;
-        if ( dflt.has_value_at_declaration() ) {
-            return &( get_descendant( dflt.value() ) );
+    if ( !child_data_->selection_default_.is_obligatory() )  { // skip for empty records
```

Comment:
Possibly introduce private method have_default_descendant in order to reduce code duplication.